src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/CustomerSegmentHandler.php

```diff
@@ -0,0 +1,266 @@
+                        $product->get_name(),
+                        $product->get_id(),
+                        $segmentPrice,
+                        $appliedPrice,
+                        $order->get_billing_email(),
+                        $quantity
+                    );
+                    wp_mail($admin_email, $subject, $message);
+                }
+            }
+        }
+    }
+
+    /**
+     * @return float|mixed
+     */
+    public function getStandardPrice($product)
+    {
+        if ($product->is_type('variable')) {
+            $defaultVariation = $product->get_available_variations()[0];
+
+            return $defaultVariation['display_price'];
+        }
+
+        return (float) $product->get_regular_price();
+    }
+
+    /**
+     * @throws WordpressException
+     */
+    public function checkPriceMismatchOnCheckout($orderId): void
+    {
+        self::checkPriceMismatchOnOrder($orderId);
+        self::saveCustomerSegmentAsOrderMeta($orderId);
+    }
+
+    public function saveCustomerSegmentAsOrderMeta($orderId): void
+    {
+        global $wpdb;
+
+        $order = wc_get_order($orderId);
+        $customerEmail = $order->get_billing_email();
+        $user = get_user_by('email', $customerEmail);
+        $userId = $user->ID;
+
+        foreach ($order->get_items() as $itemId => $item) {
+            $productId = $item->get_product_id();
+            $qty = $item->get_quantity();
+            $segmentPricesTable = CustomerSegmentPriceModel::getTableName();
+            $customerSegmentsTable = CustomerSegmentModel::getTableName();
+            $customersInSegmentsTable = CustomersInSegmentsModel::getTableName();
+            $usersTable = $wpdb->prefix . 'users';
+
+            $sql = $wpdb->prepare(
+                "SELECT * FROM 
+                    {$segmentPricesTable} sp
+                INNER JOIN 
+                    {$customerSegmentsTable} cs ON sp.customer_segment_id = cs.id
+                INNER JOIN 
+                    {$customersInSegmentsTable} cis ON cis.customer_segment_id = cs.id
+                INNER JOIN 
+                    {$usersTable} u ON cis.customer_id = u.ID
+                WHERE 
+                    sp.product_id = %d AND u.ID = %d AND sp.from_qty <= %d;
+             ",
+                $productId, $userId, $qty
+            );
+
+            $segment = $wpdb->get_row($sql);
+
+            if ($segment) {
+                update_post_meta($orderId, '_customer_segment_id', $segment->id);
+                update_post_meta($orderId, '_customer_segment_name', $segment->name);
+            } else {
+                update_post_meta($orderId, '_customer_segment_id', 'No segment found');
+                update_post_meta($orderId, '_customer_segment_name', 'No segment found');
+            }
```

Comment:
how do you know this segment was applied ? you also keep overwiting the segment on the order per product what if there 2 segments for differnet products